Update README.md and Add Module.md and Package.md files (#34)
ballerina-platform/module-ballerina-serdes: README.md

This library provides APIs for serializing and deserializing subtypes of Ballerina anydata type.

### Proto3Schema

An instance of the `serdes:Proto3Schema` class is used to serialize and deserialize ballerina values using protocol buffers.

#### Create a `serdes:Proto3Schema` object

```ballerina
// Define a type which is a subtype of anydata.
type Student record {
    int id;
    string name;
    decimal fees;
};

> // Create a serialization object by passing the typedesc.

Reviewer: ```suggestion
// Create a schema object by passing the type.
```

Author: Updated.

serdes:Proto3Schema schema = check new (Student);
```
While instantiation of this object, an underlying proto3 schema generated for the provided typedesc.

#### Serialization

```ballerina

> Student studentValue = {
>     id: 7894,
>     name: "Liam",
>     fees: 24999.99
> };
> 
> // Serialize the record value to bytes.
> byte[] serializedValue = check schema.serialize(studentValue);

Reviewer: We usually don't define variable names like `studentValue`, right?

(You can come up with a better name than `bytes`.)

```suggestion
Student student = {
    id: 7894,
    name: "Liam",
    fees: 24999.99
};

// Serialize the record value to bytes.
byte[] bytes = check schema.serialize(student);
```

Author: Updated, sticking to `bytes` for now

```

> A value having the same type as the provided typedesc can be serialized by invoking the `serialize` method on the previously instantiated `serdes:Proto3Schema` object. The underlying implementation uses the previously generated proto3 schema to serialize the provided value.

Reviewer: Let's use `type` here.

```suggestion
A value having the same type as the provided type can be serialized by invoking the `serialize` method on the previously instantiated `serdes:Proto3Schema` object. The underlying implementation uses the previously generated proto3 schema to serialize the provided value.
```

Author: Updated

> #### Serialization

Reviewer: ```suggestion
#### Deserialization
```

Author: Updated

```ballerina
Student deserializedValue = check schema.deserialize(serializedValue);

> ```

Reviewer: Let's provide a complete example for this, by creating the schema and deserializing it, considering this is a separate section.

Author: Shall we do it like this ?

```ballerina
type Student record {
    int id;
    string name;
    decimal fees;
};

byte[] bytes = readSerializedDataToByteArray();
serdes:Proto3Schema schema = check new (Student);
Student student = check schema.deserialize(bytes);
```

Author: Updated as above
